daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/XSHexBinary.scala

```diff
@@ -67,7 +67,7 @@ trait HexBinaryKind {
       case bi: JBigInt if (bi.bitLength <= 63) => reduce(bi.longValue())
       case bi: JBigInt => bi.toByteArray()
       case bd: JBigDecimal if (try { bd.toBigIntegerExact(); true } catch { case e: ArithmeticException => false }) => reduce(bd.toBigIntegerExact())
-      case str: String => reduce(new JBigInt(str))
+      case str: String => reduce(new JBigInt(str, 16))
```

Comment:
Since the spec says `dfdl:hexBinary behaves identically to the XPath 2.0 constructor function of the same name...` is there any value in making it so the string case is exactly the same as the `XSHexBinary` constructor? We can't really call its run function since we need an instance, but it just calls `StringToHexBinary.computeValue(...)`. Should we do the same? Note that that just calls the above `hexStringToByteArray` function. Maybe we should call that instead?

And to make things more confusing, we also have a `Misc.hex2Bytes` function, which seems like it has basically the same logic but throws an IllegalArgumentException's instead of NumberFormatException's. Maybe we should drop the hexStringToByteArray function in favor of Misc.hex2Byte, and change it to throw NumberFormatExceptions?

Reply:
Updated to use `Misc.hex2Bytes` and this is actually more correct since it enforces the string to have an even number of digits.